<?xml version="1.0"?>
<launch>

    <machine name="drc3" address="drc3" user="youbot" timeout="30"
                 env-loader="/home/youbot/rosenv.sh" />
    <machine name="drc1" address="drc1" user="youbot" timeout="30"
                 env-loader="/home/youbot/rosenv.sh" />

    <rosparam file="$(find assembly_common)/config/parameters.yaml" command="load" />

	<node name="fleet_master" pkg="fleet_control"
              type="fleet_master.py" output="screen" />

	<group ns="drc1">
		<node name="fleet_controller_wood" pkg="fleet_control"
                      type="fleet_controller_wood.py" output="screen" />



        <node machine="drc1" name="arm_feedback" pkg="robot"  type="arm_feedback.py" output="screen">
            <remap from="/tf" to="/drc1/tf"/>
        </node>

        <!--node name="mini_cartesian" pkg="robot"  type="mini_cartesian.py" output="screen" /-->
        <!--<node machine="drc1" name="mini_cartesian" pkg="robot"  type="mini_cartesian.py" output="log" />-->

        <!--<node name="mini_cartesian" pkg="robot"  type="mini_cartesian.py" output="screen" />-->
        <node machine="drc1" name="arm" pkg="robot"  type="arm.py" output="log" />

        <node machine="drc1" name="vicon" pkg="vicon"  type="vicon.py" output="screen" />
	</group>

	<group ns="drc3">
		<node name="fleet_controller_wood" pkg="fleet_control"
                      type="fleet_controller_wood.py" output="screen" />




        <node machine="drc3" name="arm_feedback" pkg="robot"  type="arm_feedback.py" output="screen">
            <remap from="/tf" to="/drc3/tf"/>
        </node>

        <!--node name="mini_cartesian" pkg="robot"  type="mini_cartesian.py" output="screen" /-->
        <!--<node machine="drc3" name="mini_cartesian" pkg="robot"  type="mini_cartesian.py" output="log" />-->


        <!--<node name="mini_cartesian" pkg="robot"  type="mini_cartesian.py" output="screen" />-->
        <node machine="drc3" name="arm" pkg="robot"  type="arm.py" output="log" />

        <node machine="drc3" name="vicon" pkg="vicon"  type="vicon.py" output="screen" />

	</group>

	<node name="chopsaw" pkg="fleet_control"  type="chopsaw.py" output="screen">
            <param name="port" value="/dev/ttyACM0" />
            <param name="debug" value="1" />
            <param name="oc_channel" value="saw_oc" />
            <param name="b_channel" value="saw_io" />
        </node>


</launch>
